build-logic-commons/gradle-plugin/src/main/kotlin/gradlebuild.cache-miss-monitor.gradle.kts

```diff
@@ -14,31 +14,53 @@
  * limitations under the License.
  */
 
+@file:Suppress("UNCHECKED_CAST")
+
 import com.gradle.scan.plugin.BuildScanExtension
-import java.util.concurrent.atomic.AtomicBoolean
+import org.gradle.kotlin.dsl.support.serviceOf
+
+val gradleRootProject = when {
+    project.name == "gradle" -> project.rootProject
+    project.rootProject.name == "build-logic" -> rootProject.gradle.parent!!.rootProject
+    else -> project.gradle.parent!!.rootProject
+}
 
-val cacheMissTagged = AtomicBoolean(false)
-plugins.withId("com.gradle.build-scan") {
-    val buildScan = extensions.findByType<BuildScanExtension>()
+val rootProjectName = rootProject.name
+val isInBuildLogic = rootProjectName == "build-logic"
 
-    if (System.getenv("TEAMCITY_VERSION") != null) {
-        gradle.taskGraph.afterTask {
-            if (buildCacheEnabled() && isCacheMiss() && isNotTaggedYet()) {
-                buildScan?.tag("CACHE_MISS")
+if (buildCacheEnabled() && System.getenv("TEAMCITY_VERSION") != null) {
+    gradle.taskGraph.whenReady {
+        val cacheMissMonitorBuildService: Provider<CacheMissMonitorBuildService> = gradle.sharedServices.registerIfAbsent("cacheMissMonitorBuildService-$rootProjectName", CacheMissMonitorBuildService::class) {
+            parameters.buildCacheEnabled.set(buildCacheEnabled())
+            parameters.monitoredTaskPaths.set(allTasks.filter { it.isCacheMissMonitoredTask() }.map { if (isInBuildLogic) ":build-logic${it.path}" else it.path }.toSet())
```

Comment:
I don't understand why the prefixing with `build-logic` is necessary for tasks in build logic. Is the task path here different from the one reported in the listener? What is the right method to call on the task to obtain the correct path here?

Reply:
That's what we get from the listener: `:build-logic:XX:YY`.

Reply:
Yeah, I see that you get that from the listener. Isn't there a way to get this string from the task directly without pre-fixing it with `build-logic`?